compiler/prelude/jsmapping.go

```diff
@@ -114,6 +114,9 @@ var $externalize = function(v, t) {
         return searchJsObject(v.$get(), t.elem);
       case $kindStruct:
         var f = t.fields[0];
+        if (f === undefined) {
```

Comment:
I agree that it looks better, but the defensive coder in me, that says not to trust user input, reminds me that we haven't checked that `t.fields` is an array; `t.fields.length` could be `undefined` and that does not equal 0.

We could have it check `t.fields.length > 0`, and reverse the `return` statements, although I'm not a fan of coercing `undefined`.